openedx/features/wikimedia_features/meta_translations/models.py

```diff
@@ -395,7 +396,15 @@ def create_translation_mapping(cls, base_course_blocks_data, key, value, target_
             ))
             try:
                 # For target blocks - added after rerun creation.
-                base_course_block_data = base_course_blocks_data.get(data_type=key, data=value)
+                # Check if the parent block is mapped, filter blocks based on parent_id and then compare data within those blocks
+                # Otherwise compare data throughout the course 
+                parent_translation = WikiTranslation.objects.filter(target_block__block_id=parent_id).first()
+                base_course_block_data = None
+                if parent_translation:
```

Comment:
Add log statements in both if: "Parent mapping found. Try compare data with along parent id" and else "Couldn't found parent mapping. Try just data comparison" statements.